framework/src/base/SystemBase.C

```diff
@@ -506,8 +507,11 @@ SystemBase::addVariable(const std::string & var_name,
   {
     // FIXME: we cannot refer fetype in libMesh at this point, so we will just make a copy in
     // MooseVariableBase.
-    MooseVariable * var =
-        new MooseVariable(var_num, type, *this, _subproblem.assembly(tid), _var_kind);
+    auto * var =
+        (type == (FEType(0, MONOMIAL)))
+            ? new MooseVariableConstMonomial(
+                  var_num, type, *this, _subproblem.assembly(tid), _var_kind)
+            : new MooseVariable(var_num, type, *this, _subproblem.assembly(tid), _var_kind);
```

Comment:
Holy train-wreck! You should probably split this up into a few lines. I know this was already a raw memory allocation, when you find these, see if they can be converted to smart pointers.